sdk/storage/azblob/appendblob/examples_test.go

```diff
@@ -67,3 +70,32 @@ func Example_appendblob_Client() {
 	}
 	fmt.Println(b.String())
 }
+
+// This example shows how to set an expiry time on an existing blob
+// This operation is only allowed on Hierarchical Namespace enabled accounts.
+func Example_appendblob_SetExpiry() {
+	accountName, ok := os.LookupEnv("AZURE_STORAGE_ACCOUNT_NAME")
+	if !ok {
+		panic("AZURE_STORAGE_ACCOUNT_NAME could not be found")
+	}
+	blobName := "test_append_blob_set_expiry.txt"
+	blobURL := fmt.Sprintf("https://%s.blob.core.windows.net/testcontainer/%s", accountName, blobName)
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	handleError(err)
+
+	appendBlobClient, err := appendblob.NewClient(blobURL, cred, nil)
+	handleError(err)
+
+	// set expiry on append blob to an absolute time which must be in RFC 1123 Format
```

Comment:
The wire format is an implementation detail, we shouldn't make customers do it. Can you please update the API to properly format the time?

Reply:
Given the polymorphic nature of the expressed value, I think we should use an interface instead of the const/string pairs.  Here's an example.
```go
type ExpiryType interface {
	format() (generated.ExpiryOptions, generated.BlobClientSetExpiryOptions)
}

type ExpiryTypeAbsolute time.Time

func (e ExpiryTypeAbsolute) format() (generated.ExpiryOptions, generated.BlobClientSetExpiryOptions) {
	return generated.ExpiryOptionsAbsolute, generated.BlobClientSetExpiryOptions{
		ExpiresOn: to.Ptr(time.Time(e).UTC().Format(http.TimeFormat)),
	}
}

type ExpiryTypeRelativeToNow time.Duration

func (e ExpiryTypeRelativeToNow) format() (generated.ExpiryOptions, generated.BlobClientSetExpiryOptions) {
	return generated.ExpiryOptionsRelativeToNow, generated.BlobClientSetExpiryOptions{
		ExpiresOn: to.Ptr(strconv.FormatInt(time.Duration(e).Milliseconds(), 10)),
	}
}
```
This lets callers naturally specify the desired time, and the conversion is performed by the underlying types.
```go
func (b *Client) SetExpiry(ctx context.Context, expiryType ExpiryType, o *SetExpiryOptions) (SetExpiryResponse, error) {
	opt, val := expiryType.format()
	resp, err := b.generated().SetExpiry(ctx, opt, &val)
	return resp, err
}
...
blobClient.SetExpiry(context.TODO(), blob.ExpiryTypeAbsolute(time.Now().Add(12*time.Hour)), nil)

blobClient.SetExpiry(context.TODO(), blob.ExpiryTypeRelativeToNow(10*time.Second), nil)
```

Reply:
Made the changes. Please review.